libnmstate/nm/sriov.py

```diff
@@ -17,16 +17,63 @@
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
 
-from libnmstate.error import NmstateNotImplementedError
+from libnmstate.error import NmstateNotSupportedError
+from libnmstate.nm import connection as nm_connection
+from libnmstate.nm import device
+from libnmstate.nm import nmclient
 from libnmstate.schema import Ethernet
+from libnmstate.schema import Interface
 
 
-def create_setting(iface_state):
-    setting = None
+def create_setting(iface_state, base_con_profile):
+    sriov_setting = None
+    ifname = iface_state[Interface.NAME]
     sriov_config = iface_state.get(Ethernet.CONFIG_SUBTREE, {}).get(
         Ethernet.SRIOV_SUBTREE
     )
     if sriov_config:
-        raise NmstateNotImplementedError
+        if not _has_sriov_capability(ifname):
+            raise NmstateNotSupportedError(
+                f"Interface '{ifname}' does not support SR-IOV"
+            )
 
-    return setting
+        sriov_setting = base_con_profile.get_setting_duplicate(
+            nmclient.NM.SETTING_SRIOV_SETTING_NAME
+        )
+        if not sriov_setting:
+            sriov_setting = nmclient.NM.SettingSriov.new()
+
+        sriov_setting.props.total_vfs = sriov_config[Ethernet.SRIOV.TOTAL_VFS]
+
+    return sriov_setting
+
+
+def _has_sriov_capability(ifname):
+    dev = device.get_device_by_name(ifname)
+    if nmclient.NM.DeviceCapabilities.SRIOV & dev.props.capabilities:
+        return True
+
+    return False
+
+
+def get_info(device):
+    """
+    Provide the current active SR-IOV values for a device
```

Comment:
The docstring is a little bit misleading since it is reporting the configuration and not the values from the device. On the other hand, we assume that the live configuration is the actual devices settings. What do you think?

Reply:
I agree. Done, thanks!